### PR TITLE
f32/f64: add Affine (alpha*x+beta) and Log10Floored fused kernels

### LINKED ISSUE #298: f32/f64: add Affine (alpha*x + beta) scalar-affine kernel
## Summary

Add a scalar-affine elementwise kernel computing `dst[i] = alpha*src[i] + beta`, where `alpha` and `beta` are scalars, for f32 and f64 for parity. This is the fusion of the existing `Scale` (alpha*x) and `AddScalar` (x + beta) into a single pass, and sits naturally alongside `Scale`, `AddScalar`, `AddScaled` (dst += alpha*s), and `FMA` (a*b + c, all vectors).

## Motivation

Consumer: tphakala/go-audio-dsp#42. The spectrogram dB stage in `spectrogram/config.go` runs the dB-conversion tail as `Scale(dst, dst, 10)` followed by `AddScalar(dst, dst, dbAddConst)`, two full-array passes over the selected bins that together are one affine map `y = 10*x + dbAddConst`. A single affine kernel computes it in one pass. The operation carries no DSP semantics (normalization, quantize/dequantize, and unit conversion all want the same `alpha*x + beta`), so it belongs here as a generic primitive rather than being vendored as a domain kernel in go-audio-dsp.

## Proposed API

```go
// Affine computes dst[i] = alpha*src[i] + beta for every element.
// In-place safe (dst may alias src), same length contract as Scale.
func Affine(dst, src []float32, alpha, beta float32)
```

Plus the f64 counterpart. The name is a suggestion; `ScaleAddScalar` would also read consistently with the existing `AddScalar` and `AddScaled`.

## Numeric contract question

Two options, your call:

- Split rounding (`alpha*x`, then `+ beta`, two roundings): bit-identical to the current `Scale`-then-`AddScalar` sequence, so consumers observe no change.
- FMA-based (single rounding, `fma(alpha, x, beta)`): faster and the natural hardware form, but tolerance-stable rather than bit-identical, matching the existing FMA-family contract.

For the go-audio-dsp#42 consumer the dB output tolerance across SIMD tiers is already an accepted open question, so either contract works there. I lean toward the FMA-based form for the performance win and consistency with the FMA family, but I will follow whatever contract you decide.

### LINKED ISSUE #299: f32/f64: add Log10Floored (log10 of a floored input) fused kernel
## Summary

Add a fused floor-then-log10 kernel for f32 and f64: `dst[i] = log10(max(src[i], floor))`. This folds the common `Clamp(x, floor, +Inf)` then `Log10` pair into a single pass and guarantees a finite result (no `log10(0) = -Inf`) when the input can be zero.

## Motivation

Consumer: tphakala/go-audio-dsp#42 and the spectrogram dB stage generally. Taking log10 of magnitude or power data almost always requires flooring the input first so silence does not map to -Inf, so today it is two full-array passes (`Clamp` then `Log10`). The pattern is generic: any log-magnitude or log-probability transform over data that can be zero wants the same floored log. Combined with the affine kernel proposed in simd#298, the spectrogram dB tail `10*log10(max(power, floor)) + c` drops from three passes to two.

## Proposed API

```go
// Log10Floored computes dst[i] = log10(max(src[i], floor)) for every element.
// The floor keeps the result finite when src can be zero.
// In-place safe (dst may alias src).
func Log10Floored(dst, src []float32, floor float32)
```

Plus the f64 counterpart. A two-sided `Log10Clamped(dst, src, lo, hi)` mirroring `Clamp` would also be reasonable if you prefer symmetry; the spectrogram only needs the lower floor.

## Optional: full log-affine fusion

If you would rather collapse the whole tail in one kernel, a decibel-style `dst[i] = alpha*log10(max(src[i], floor)) + beta` (folding simd#298's affine into this one) takes the spectrogram dB path from three passes to one. That is more specialized than a plain floored log, so I lean toward shipping the generic `Log10Floored` and letting the affine from simd#298 compose with it, but I will take whichever you prefer.

## Priority

Low. This is a pass-count optimization, not a correctness fix. Per the PR #293 profiling lesson (the MulAdd fusion turned out to be ~2% of ISTFT runtime), it is worth a benchmark on the spectrogram dB path before implementing. I am filing it now so the dependency is tracked rather than left implicit in the downstream issue.

## Numeric contract

The floor is an exact `max`, so `Log10Floored` is bit-identical to the current `Clamp`-then-`Log10` sequence. The optional log-affine variant inherits the affine numeric-contract choice discussed in simd#298.

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Adds two fused elementwise kernels to the f32 and f64 packages, covering both #298 (Affine) and #299 (Log10Floored).

Affine computes `dst[i] = alpha*src[i] + beta` in a single pass, the fused form of Scale followed by AddScalar. It uses split rounding (a separate multiply then add, two IEEE-754 roundings), so it is bit-identical to Scale then AddScalar on every dispatch tier and never contracts into a fused multiply-add. There are new kernels affineAVX/AVX512/SSE(2) and affineNEON for each type, plus a pure-Go fallback. The f32 fallback uses a `float32()` barrier and the f64 fallback uses two passes to defeat the compiler's FMADD contraction on FMA-capable backends; the single-expression `a*x+b` form fuses to one rounding on arm64, which would break the split-rounding contract. All four f32 and four f64 affine kernels are registered in `singleRoundingKernels`, so `TestNoFMAContract` statically forbids fusing any of them regardless of the host arch.

Log10Floored computes `dst[i] = log10(max(src[i], floor))`, the floored base-10 logarithm, guaranteeing a finite result for a positive floor (silence maps to `log10(floor)` instead of `-Inf`). It composes the existing clamp and log10 kernels (clamp to `[floor, +Inf]` then log10), bit-identical to that pair on each tier. It is the drop-in replacement for the common Clamp-then-Log10 pair in a spectrogram dB stage.

Both are zero-allocation and in-place safe (dst may alias src).

## Related Issues

Fixes #298
Fixes #299

## Test Plan

- [x] `go test ./...` green on amd64 across all dispatch tiers (tier sweep)
- [x] f32 and f64 suites green on native arm64 (Raspberry Pi 5, Cortex-A76), full set with testdata
- [x] Bit-exact parity for Affine vs Scale-then-AddScalar and vs a scalar reference, across every tier, ragged sizes, and in-place aliasing
- [x] Bit-exact Log10Floored vs Clamp-then-Log10, plus a finite-result check over zeros and negatives
- [x] Differential and finiteness fuzzing (Affine and Log10Floored, f32 and f64), passing on both amd64 and arm64 over NaN/Inf/subnormal inputs
- [x] `AllocsPerRun == 0` on every tier
- [x] asmcheck: WORD-encoding, ISA-gate, FMA3-gate, and TestNoFMAContract (now covering the f64 affine kernels) all pass
- [x] `golangci-lint run ./...` clean

## Summary by CodeRabbit

- **New Features**
  - Added `Affine` operations for float32 and float64 slices, calculating a scaled value plus an offset.
  - Added `Log10Floored`, which applies a lower bound before calculating base-10 logarithms.
  - Both operations support in-place processing, documented rounding behavior, and portable fallbacks.
  - Optimized implementations are available across supported CPU architectures.

- **Tests**
  - Added comprehensive edge-case, aliasing, allocation, benchmark, and fuzz coverage, including special floating-point values.

### CODE REVIEW

File: f32/f32.go [hunk 1]
```diff
@@ -1019,6 +1039,27 @@ func Log10(dst, src []float32) {
...
+func Log10Floored(dst, src []float32, floor float32) {
+	n := min(len(dst), len(src))
+	if n == 0 {
+		return
+	}
+	// Exact lower floor (max with floor via a +Inf upper bound), then the
+	// standard log10 kernel. For the intended positive-finite magnitude/power
+	// domain the clamp introduces no rounding beyond Log10's own.
+	clamp32(dst[:n], src[:n], floor, float32(math.Inf(1)))
+	log10_32(dst[:n], dst[:n])
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32.go:1063]: Corrected the docs to stop calling this fused: Log10Floored composes an exact lower clamp with the existing log10 kernel (two passes). I'm deferring the true single-pass floored-log10 rather than folding a floor into the shared log core: the log polynomial dominates (the clamp pass measured ~8% of Log10Floored's runtime), that core is shared by log/log2/log10/pow under the #156 rounding contract with asmcheck WORD verification, and #299 itself rates the fusion low-priority pending a benchmark. Filing a follow-up to track the single-pass kernel. Framing fixed in 2201dbe.

File: f32/f32.go [hunk 2]
```diff
@@ -1019,6 +1039,27 @@ func Log10(dst, src []float32) {
...
 
+// Log10Floored computes dst[i] = log10(max(src[i], floor)) for every element: a
+// floored base-10 logarithm. The floor is an exact lower clamp applied before the
+// logarithm, so a zero, negative, or tiny input maps to log10(floor) rather than
+// -Inf or NaN; pass a small positive floor (for example a spectrogram noise
+// floor) to keep every result finite.
+//
+// It is the fused form of Clamp(dst, src, floor, +Inf) followed by
+// Log10(dst, dst) and is bit-identical to that pair on each dispatch path.
+// Processes min(len(dst), len(src)) elements; in-place safe (dst may alias src).
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32.go]: Fixed in 2201dbe. The doc now says Log10Floored composes clamp + the existing log10 kernel in two passes, not a single fused kernel.

File: f64/f64.go [hunk 3]
```diff
@@ -152,6 +162,22 @@ func AddScalar(dst, a []float64, s float64) {
 	addScalar(dst[:n], a[:n], s)
 }
 
+// Affine computes the scalar-affine map dst[i] = alpha*src[i] + beta for every
+// element: the fused, single-pass form of Scale (alpha*x) followed by AddScalar
+// (x + beta). The multiply and the add round separately (two IEEE-754 roundings),
+// so the result is bit-identical to Scale(dst, src, alpha) then
+// AddScalar(dst, dst, beta) on every dispatch path; the kernels never contract
+// into a fused multiply-add. Where a single fused rounding is wanted, use FMA.
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f64/f64.go]: Fixed in 2201dbe. Dropped 'single-pass' from the Affine doc. The f64 pure-Go fallback deliberately uses two passes to stop the compiler contracting a*x+b into one FMADD on arm64, so 'fused form of Scale then AddScalar' is the accurate description across all tiers.

File: f64/f64.go [hunk 4]
```diff
@@ -902,6 +928,27 @@ func Log10(dst, src []float64) {
...
 
+// Log10Floored computes dst[i] = log10(max(src[i], floor)) for every element: a
+// floored base-10 logarithm. The floor is an exact lower clamp applied before the
+// logarithm, so a zero, negative, or tiny input maps to log10(floor) rather than
+// -Inf or NaN; pass a small positive floor (for example a spectrogram noise
+// floor) to keep every result finite.
+//
+// It is the fused form of Clamp(dst, src, floor, +Inf) followed by
+// Log10(dst, dst) and is bit-identical to that pair on each dispatch path.
+// Processes min(len(dst), len(src)) elements; in-place safe (dst may alias src).
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f64/f64.go]: Fixed in 2201dbe, same as the f32 doc: it now describes the two-pass clamp+log10 composition rather than a fused kernel.
